### DISCUSSION TITLE
ESP32-C3 USB console

### DISCUSSION THREAD

**Discussion by Zambiorix** [espressif/arduino-esp32#6871] [Q&A] [answered]
Hi,

I have a custom development board with a ESP32-C3 and a USB connector directly connected to pins 18 & 19.
The default UART pins are used for another purpose.

I have a **VSCode** / **PlatformIO** / **Arduino** project.

Uploading works fine.

But how do I get the console output?

```
    Serial.begin(460800); 

    Serial.println("hello world");
```

Doesn't return anything ...

Thanks

**Comment by me-no-dev** [comment]
You need to enable CDC On Boot for `Serial` to be mapped to the USB port

**Reply by Zambiorix** [reply]
Thank you,

And how exactly do I do that on my VSCode / PlatformIO / Arduino project?
I have tried many variations of setting build flags in my platformio.ini file, but nothing seems to work.

**Reply by me-no-dev** [reply]
@Jason2866 do you know what flag needs to be added to PIO?

**Reply by Jason2866** [reply]
This platformio boards.json should work https://github.com/platformio/platform-espressif32/blob/develop/boards/adafruit_qtpy_esp32c3.json since it is a board which uses CDC too

**Reply by Zambiorix** [reply]
Thank you

**Comment by Zambiorix** [comment, marked as answer]
For those with a `esp32-c3-devkitm-1` board, or with a custom board, this does the trick:

```
board = esp32-c3-devkitm-1
monitor_speed = 460800
build_flags = 
	-D ARDUINO_USB_MODE=1
	-D ARDUINO_USB_CDC_ON_BOOT=1
```

**Reply by BeeGeeBee** [reply]
This worked a treat with ESP32-C3 Super Mini just arrived from AliExpress.  @Zambiorix

**Reply by MainWire** [reply]
Fix confirmed for ESP32-C3 Super Mini from Aliexpress ✔️

**Reply by myhd** [reply]
Same here. Bang!

**Reply by jacekgajek** [reply]
Worked for ESP32-C3-Zero-M by Waveshare 25532. THANK YOU!!!
